lib/download.bash

```diff
@@ -41,6 +41,28 @@ need_cmd() {
   fi
 }
 
+# This function retires the download - if it fails due to network issues.
+retry_download(){
+  local max_retries=5
+  local retry_delay=2
+
+  for (( i=0 ; i<max_retries ; i++ ))
+  do
+     if [ "$3" = "curl" ]; then
+        curl -sSfL "$1" -o "$2" && return 0
+     elif [ "$3" = "wget" ]; then
+        wget "$1" -O "$2"
+     fi
```

Comment:
Overall I'm curious that one of these has explicit return on success and the other doesn't. If there's a behavioural difference between the two commands this would be great to document, otherwise let's make it consistent.